porting-a-new-endpoint.md

```diff
@@ -0,0 +1,201 @@
+    ...
+	"DeleteTag_RunId":                    "required",
+	"DeleteTag_Key":                      "required",
+}
+```
+
+After that, run `mage generate` again and check if our fields in `DeleteTag` now contain `validate:"required"`.
+
+## Implement the Service Method
+
+We aim to keep the Go implementation as close to the Python code as possible. That's why we have a thin service layer and keep most logic in the (SQL) store instead of the service. This keeps things similar to Python, making it easier to port and compare the code.
+
+In the service, we need to call the store with the same arguments as the Python store would be called. This is typically where we convert proto structs to entities. In the case of `deleteTag`, we can just pass the two arguments to the store.
+
+```go
+func (ts TrackingService) DeleteTag(ctx context.Context, input *protos.DeleteTag) (*protos.DeleteTag_Response, *contract.Error) {
+    // Note that Store.DeleteTag does not exist yet; this just mirrors what happens in _delete_tag() in handlers.py
+	if err := ts.Store.DeleteTag(ctx, input.GetRunId(), input.GetKey()); err != nil {
+		return nil, err
+	}
+
+	return &protos.DeleteTag_Response{}, nil
+}
+```
+
+## Expose the Endpoint via FFI
+
+To access our new endpoint in the Python FFI binding, we need to update our Python store as well. Update [mlflow_go/store/tracking.py](mlflow_go/store/tracking.py) and add the method matching our endpoint from [.mlflow.repo/mlflow/store/tracking/sqlalchemy_store.py](.mlflow.repo/mlflow/store/tracking/sqlalchemy_store.py):
+
+```python
+def delete_tag(self, run_id, key):
+	request = DeleteTag(run_id=run_id, key=key)
+	self.service.call_endpoint(get_lib().TrackingServiceDeleteTag, request)
+```
+
+Note that `DeleteTag` and `TrackingServiceDeleteTag` will match our newly generated code in [pkg/lib/tracking.g.go](pkg/lib/tracking.g.go).
+
+## Store Implementation
+
+In the future, we hope to implement the file store in Go as well. That’s why we use a store interface to abstract away all SQL details, which also makes testing easier.
+
+Update [store.go](pkg/tracking/store/store.go):
+
+```go
+	RunTrackingStore interface {
+		...
+		DeleteTag(ctx context.Context, runID, key string) *contract.Error
+	}
+```
+
+Then, add an empty implementation in [store/sql/tags.go](pkg/tracking/store/sql/tags.go):
+
+```go
+func (s TrackingSQLStore) DeleteTag(
+	ctx context.Context, runID string, key string,
+) *contract.Error {
+	return nil
+}
+```
+
+Lastly, update the `TrackingStore` interface mock via `go generate ./...`.
+
+If everything goes well, you should be able to run `mage test:unit` to execute the Go unit tests. This serves as a sanity check to verify that the Go code is compiling.
+
+### The Actual Store Implementation
+
+At this point, we need to write the actual implementation. Again, we want to stay true to the Python code, so please refer to [sqlalchemy_store.py](.mlflow.repo/mlflow/store/tracking/sqlalchemy_store.py).
```

Comment:
Just a minor issue: clicking the link doesn't seem to take us to the file since it isn't actually part of the current repo. Can we perhaps link the file from an external repo, or simply show the file path?